x-pack/plugins/lens/public/shared_components/axis/extent/helpers.ts

```diff
@@ -86,9 +103,39 @@ export function getDataBounds(
   }
 }
 
-export function validateExtent(shouldIncludeZero: boolean, extent?: UnifiedAxisExtentConfig) {
-  return {
-    inclusiveZeroError: shouldIncludeZero && !validateZeroInclusivityExtent(extent),
-    boundaryError: !validateAxisDomain(extent),
-  };
+export function validateExtent(
+  hasBarOrArea: boolean,
+  extent: UnifiedAxisExtentConfig,
+  scaleType?: YScaleType | XScaleType
+): {
+  helpMsg?: string;
+  errorMsg?: string;
+} {
+  if (!validateAxisDomain(extent)) {
+    return {
+      errorMsg: i18n.translate('xpack.lens.axisExtent.boundaryError', {
+        defaultMessage: 'Lower bound has to be less than upper bound',
+      }),
+    };
+  }
+
+  if (scaleType === 'log') {
+    const key = validateLogarithmicExtent(extent) ? 'helpMsg' : 'errorMsg';
+    return {
+      [key]: i18n.translate('xpack.lens.axisExtent.logExcludeZero', {
+        defaultMessage: 'Logarithmic bounds must exclude zero',
```

Comment:
Similar to the previous message, it's a very technical langauge. I'd prefer something like 'Logarithmic bounds cannot be equal to 0' CC @amyjtechwriter

Reply:
Yeah, this came from the inverse of the linear case above. But we would have to focus the message to either the lower or upper like you suggested above because a range does not make sense to equal `0`.

Reply:
I just tested it and I noticed that 'must include zero' means that one of the bound has to equal zero, while `must exclude zero` means that zero cannot be within the range. That makes it more nuanced 🙈

Reply:
> I just tested it and I noticed that 'must include zero' means that one of the bound has to equal zero

No this is not true, the lower bound can be negative so long as the upper bound is greater...

![Zight Recording 2024-04-29 at 09 23 55 PM](https://github.com/elastic/kibana/assets/19007109/728c7d63-f981-47f1-bc07-d9c21910ac7d)

I feel like the idea is being overcomplicated, in it's simplest form the custom bounds is defining a numerical [interval](https://en.wikipedia.org/wiki/Interval_(mathematics)) from a min to a max value that must include or exclude `0`.

The idea you had about isolating the problematic endpoint (i.e. min or max) could have a better UX. But this is somewhat fixable if the user changes the scale or switches to `full`/`data` mode and back to `custom`, the issue will resolve with the correct bounds.

Reply:
All of this becomes more clear to me if we use "contains/should not contain" instead of exclude.

"Logarithmic bounds must not contain zero"

Marta's suggestion is even better if we can split it.

Reply:
Thanks @timductive and @mbondyra I updated all the copy to the suggestions.

I'll look into splitting the field messaging to be independent of each other, I think that would be the best case and we can also call out missing bounds.